File: app/src/main/java/com/tolstoy/zurichat/ui/activities/OpenCameraActivity.java
package com.tolstoy.zurichat.ui.activities;

import androidx.activity.result.ActivityResult;
import androidx.activity.result.ActivityResultCallback;
import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.provider.MediaStore;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.Toast;

import com.tolstoy.zurichat.R;

public class OpenCameraActivity extends AppCompatActivity {

    public static final int CAMERA_ACTION_CODE = 1;
    ImageView imageView;
    ImageView imageButton;
    ActivityResultLauncher<Intent> activityResultLauncher;
    Boolean isImageFitToscreen = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_open_camera);

        imageView = findViewById(R.id.displayid_ImgV);
        imageButton = findViewById(R.id.camera_btn);

        // this function accesses the camera, converts image to bitmap and displays it.
        activityResultLauncher = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
            if (result.getResultCode() == RESULT_OK && result.getData() != null) {
                Bundle bundle = result.getData().getExtras();
                Bitmap bitmap = (Bitmap) bundle.get("data");
                imageView.setImageBitmap(bitmap);

            }
        });

        // click listener for the camera button to start the camera.
        imageButton.setOnClickListener(view -> {
            Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
            if (intent.resolveActivity(getPackageManager()) != null){
                activityResultLauncher.launch(intent);
            }else{
                Toast.makeText(OpenCameraActivity.this, "Unable to Start Camera!",
                        Toast.LENGTH_SHORT).show();
            }
        });
//        imageView.setOnClickListener(view -> {
//            if (isImageFitToscreen) {
//                imageView.setLayoutParams(new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams
//                        .WRAP_CONTENT, RelativeLayout.LayoutParams.WRAP_CONTENT));
//                imageView.setAdjustViewBounds(true);
//            }else {
//                isImageFitToscreen = true;
//                imageView.setLayoutParams(new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams
//                        .MATCH_PARENT,RelativeLayout.LayoutParams.MATCH_PARENT));
//                imageView.setScaleType(ImageView.ScaleType.FIT_XY);
//
//            }
//        });

        }

    }

File: app/src/main/java/com/tolstoy/zurichat/ui/activities/OpenCameraActivity.kt
package com.tolstoy.zurichat.ui.activities

import android.app.Activity
import android.content.Intent
import android.content.pm.PackageManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.provider.MediaStore
import android.provider.SyncStateContract
import android.widget.ImageView
import android.widget.Toast
import androidx.activity.result.ActivityResult
import androidx.activity.result.ActivityResultCallback
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts
import androidx.core.app.ActivityCompat
import com.tolstoy.zurichat.R
import java.util.jar.Manifest


class OpenCameraActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_open_camera)
        val cameraButton = findViewById<ImageView>(R.id.camera_btn)

        cameraButton.setOnClickListener{
            if (ActivityCompat.checkSelfPermission(this, android.Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
                ActivityCompat.requestPermissions(
                    this,
                    arrayOf(android.Manifest.permission.CAMERA),
                    111
                )
                Toast.makeText(applicationContext, "Unable to start camera", Toast.LENGTH_SHORT).show()
            } else
                Toast.makeText(applicationContext, "Success!", Toast.LENGTH_SHORT).show()

            var i = Intent(MediaStore.ACTION_IMAGE_CAPTURE)
            startActivityForResult(i,101)
        }

    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<out String>,
        grantResults: IntArray
    ) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        if(requestCode == 111 && grantResults[0] == PackageManager.PERMISSION_GRANTED){
            Toast.makeText(applicationContext, "Success!", Toast.LENGTH_SHORT).show()
        }
    }


}